src/main/java/org/jenkinsci/remoting/protocol/impl/BIONetworkLayer.java

```diff
@@ -248,10 +249,23 @@ public void run() {
                         } catch (ClosedChannelException e) {
                             onRecvClosed();
                             return;
-                        } catch (EOFException e) {
+                        } catch (IOException e) {
```

Comment:
Not sure you want to catch `java.net.SocketTimeoutException` which would be thrown for a read timeout but is non fatal...  [javadoc link](https://docs.oracle.com/javase/7/docs/api/java/net/Socket.html#setSoTimeout(int))

Not sure if the timeout is ever set here - but still - was the cause of a previous bug in remoting...